go/cmd/dolt/commands/reset.go

```diff
@@ -80,90 +83,111 @@ func (cmd ResetCmd) ArgParser() *argparser.ArgParser {
 		}
-
-		return handleResetSoftTables(ctx, apr.Args, roots, dEnv, usage)
 	}
-}
 
-func handleResetSoftTables(ctx context.Context, tables []string, roots doltdb.Roots, dEnv *env.DoltEnv, usage cli.UsagePrinter) int {
-	roots, err := actions.ResetSoft(ctx, dEnv.DbData(), tables, roots)
+	// process query through prepared statement to prevent sql injection
+	query, params := constructParametrizedDoltResetQuery(apr)
+	interpolatedQuery, err := dbr.InterpolateForDialect(query, params, dialect.MySQL)
 	if err != nil {
-		return handleResetError(err, usage)
+		cli.Println(err.Error())
+		return 1
 	}
-
-	err = dEnv.UpdateRoots(ctx, roots)
+	_, _, err = queryist.Query(sqlCtx, interpolatedQuery)
 	if err != nil {
-		return handleResetError(err, usage)
+		cli.Println(err.Error())
+		return 1
 	}
 
-	printNotStaged(ctx, dEnv, roots.Staged)
-	return 0
-}
-
-func handleResetSoftToRef(ctx context.Context, dEnv *env.DoltEnv, ref string, usage cli.UsagePrinter) int {
-	newRoots, err := actions.ResetSoftToRef(ctx, dEnv.DbData(), ref)
-	if err != nil {
-		return handleResetError(err, usage)
+	if !apr.Contains(HardResetParam) {
+		printNotStaged(sqlCtx, queryist)
 	}
 
-	err = dEnv.UpdateStagedRoot(ctx, newRoots.Staged)
-	return handleResetError(err, usage)
+	return 0
 }
 
-func handleResetHard(ctx context.Context, apr *argparser.ArgParseResults, usage cli.UsagePrinter, dEnv *env.DoltEnv, roots doltdb.Roots) int {
-	arg := ""
-	if apr.NArg() > 1 {
-		return handleResetError(fmt.Errorf("--hard supports at most one additional param"), usage)
-	} else if apr.NArg() == 1 {
-		arg = apr.Arg(0)
-	}
+// constructParametrizedDoltResetQuery generates the sql query necessary to call the DOLT_RESET() stored procedure with placeholders
+// for arg input. Also returns a list of the inputs in the order in which they appear in the query.
+func constructParametrizedDoltResetQuery(apr *argparser.ArgParseResults) (string, []interface{}) {
+	var params []interface{}
+	var param bool
 
-	headRef, err := dEnv.RepoStateReader().CWBHeadRef()
-	if err != nil {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
-	}
-	ws, err := dEnv.WorkingSet(ctx)
-	if err != nil {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+	var buffer bytes.Buffer
+	var first bool
+	first = true
+	buffer.WriteString("CALL DOLT_RESET(")
+
+	writeToBuffer := func(s string) {
+		if !first {
+			buffer.WriteString(", ")
+		}
+		if !param {
+			buffer.WriteString("'")
+		}
+		buffer.WriteString(s)
+		if !param {
+			buffer.WriteString("'")
+		}
+		first = false
+		param = false
 	}
 
-	err = actions.ResetHard(ctx, dEnv, arg, roots, headRef, ws)
-	if err != nil {
-		return handleResetError(err, usage)
+	if apr.Contains(HardResetParam) {
+		writeToBuffer("--hard")
+		if apr.NArg() == 1 {
+			param = true
+			writeToBuffer("?")
+			params = append(params, apr.Arg(0))
+		}
+	} else if apr.Contains(SoftResetParam) {
+		writeToBuffer("--soft")
+		if apr.NArg() > 0 {
```

Comment:
According to the docs, soft takes no more than one arg. Take the loop out of this.